### PR TITLE
Differentiate between TNeededProps and TChildProps in mapProps

### PR DESCRIPTION
**Pull request by ianks**
This is a re-do of https://github.com/evolution-gaming/typed-recompose/pull/3

This one includes tests :smile:

### CODE REVIEW

File: test/test.tsx [hunk 1]
```diff
@@ -52,19 +55,29 @@ test('mapProps', t => {
...
+    interface NeededProps {
+        firstName: string;
+        lastName: string;
+    }
+
+    interface MappedProps {
+        fullName: string;
+    }
+
+    class FullName extends React.Component<MappedProps, {}> {
```
Comment (pablobirukov) [inline test/test.tsx]: Minor, but `{}` should be `void`

Comment (ianks) [reply test/test.tsx]: Changed.

File: test/test.tsx [hunk 2]
```diff
@@ -1,3 +1,6 @@
+/// <reference path='./../typings/index.d.ts' />
```
Comment (pablobirukov) [inline test/test.tsx:1]: Why do we need this?

Comment (pablobirukov) [reply test/test.tsx:1]: I mean which problem are you trying to solve? I think it should be possible to solve it using `tsconfig.json`

**Comment by ianks** [comment]
@r00ger can this be merged?

**Comment by pablobirukov** [comment]
@ianks sure thing I just asked why have you added `/// <reference path='./../typings/index.d.ts' />`
Please take a look at code review. Didn't you get email notification because I didn't mention you?

**Review by pablobirukov — APPROVED** [review]

File: test/test.tsx (hunk 2 above)
Comment (ianks) [reply test/test.tsx:1]: In my editor is says that it cannot find module `recompose`

**Pull request merged**